Nodejs/Product/Nodejs/Project/NodejsProjectNode.cs

```diff
@@ -202,7 +201,11 @@ public NodejsProjectNode(NodejsProjectPackage package)
 
         public bool IsTypeScriptProject {
             get {
```

Comment:
Does the race condition still exist here? Except now, we might not download typings to some JS projects? Or is this why you're making use of the UI thread?